express-notes-api/index.js

```diff
@@ -0,0 +1,87 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json')
+const notes = data.notes;
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const x in notes){
+    arr.push(notes[x]);
+  }
+  res.status(200).json(arr);
+})
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  if(id <= 0){
+    const error400 = {error: "id must be a positive integer"}
+    res.status(400).json(error400);
+  }
+  else if(id in notes) {
+    res.json(notes[id]);
+  }
+  else{
+    const error404 = {error: `cannot find note with id ${id}`};
+    res.status(404).json(error404);
+  }
+})
+
+app.post('/api/notes', (req, res) => {
+  if(req.body.content.length == 0){
+    res.status(400).json({ "error": "content is a required field"})
+  }
+  else if (req.body.content.length > 0){
+    const newNote = {
+      id: data.nextId,
+      content: req.body.content
+    };
+    notes[data.nextId] = newNote;
+    data.nextId++;
```

Comment:
Once the new note has been added to `notes` you need to write the data back to the `data.json` file